### DISCUSSION TITLE
Setup for flutter development?

### DISCUSSION THREAD

**Discussion by Quoteme** [LazyVim/LazyVim#765] [Q&A]
Hi guys, I found this Reddit post about flutter-tools inside lazyvim: https://www.reddit.com/r/neovim/comments/11wjonj/how_to_make_fluttertools_work_on_lazyvim/

The problem is, that code actions, which are heavily relied on when programming in flutter, are not available when pressing "<leader>ca".

Can any of you suggest a good flutter-development setup?

**Comment by Quoteme** [comment]
Great news! I found a solution:

Create a new file `flutter.lua` under `.config/nvim/lua/plugins/flutter.lua` and put this inside:

```
return {
  {
    "akinsho/flutter-tools.nvim",
    lazy = false,
    dependencies = {
      "nvim-lua/plenary.nvim",
      "stevearc/dressing.nvim", -- optional for vim.ui.select
    },
    config = function()
      require("flutter-tools").setup({})
    end,
  },
}
```

also, I used [direnv](https://direnv.net/) and had two different locations where the fluttersdk were installed. Disableing direnv in my projects directory made everything work.

**Reply by amirhosseinghanipour** [reply]
Also, if anyone's using `stylua` or `shfmt`, be aware that they overwrite your dartls formatting no matter if you force `format_on_save` as it might break LazyVim.
